src/AuthContext.tsx

```diff
@@ -6,7 +6,16 @@ import {
   AuthContextProps,
 } from './AuthContextInterface';
 
-export const AuthContext = React.createContext<AuthContextProps | undefined>(undefined);
+export const defaultAuthContext: AuthContextProps = {
+  signIn: () => Promise.reject(new Error('AuthContext not loaded yet.')),
+  signInPopup: () => Promise.reject(new Error('AuthContext not loaded yet.')),
+  signOut: () => Promise.reject(new Error('AuthContext not loaded yet.')),
+  signOutRedirect: () => Promise.reject(new Error('AuthContext not loaded yet.')),
```

Comment:
Would these ever hit `Promise.reject`? Are we only doing this since Typescript requires it?

Reply:
In theory this should never bit hit because `useAuth` will already throw an exception if it's hit. The only reasons for it were Typescript and also to clearly indicate that if something else was missed something had gone wrong and it wasn't loaded.

Reply:
Nice, like it. We could probably do something like this too?

```suggestion
  signIn: notLoaded,
  signInPopup: notLoaded,
  signOut: notLoaded,
  signOutRedirect: notLoaded,
```

Setting it like this:

```typescript
const notLoaded = () => Promise.reject(new Error('AuthContext not loaded yet.'));
```

Does it make sense to do something like this? (not tested).

```
const notLoaded = () => throw new Error('AuthContext not loaded yet.');
```

Reply:
Oh yeah, I like that! I can play around with trying a few different combos and get back to you.